db/version_set.cc

```diff
@@ -3240,11 +3242,19 @@ Status VersionSet::ApplyOneVersionEdit(
           edit.column_family_name_);
     }
     auto cf_options = name_to_options.find(edit.column_family_name_);
-    if (cf_options == name_to_options.end()) {
+    // implicitly add persistent_stats column family without requiring user
+    // to specify
+    bool isPersistStatsColumnFamily = edit.column_family_name_.compare(
+        kPersistentStatsColumnFamilyName) == 0;
+    if (cf_options == name_to_options.end() && !isPersistStatsColumnFamily) {
       column_families_not_found.insert(
           {edit.column_family_, edit.column_family_name_});
     } else {
-      cfd = CreateColumnFamily(cf_options->second, &edit);
+      if (isPersistStatsColumnFamily) {
+        cfd = CreateColumnFamily(ColumnFamilyOptions(), &edit);
```

Comment:
Creating this column family options in this way may not be good for stats. For example, it uses a 64MB memtable, and create a new 8MB block cache. We should be careful about the resource this CF uses.

Reply:
since we are hiding persistent stats column family from users, user will not be able to specify CFOptions for this CF, maybe we should have a LITE version of CFOptions?